### ISSUE TITLE
no-unused-vars only working for certain parameters

### ISSUE THREAD

**Issue by armandAkop** [eslint/eslint#14169]
**Tell us about your environment**

* **ESLint Version:** 7.21.0
* **Node Version:** 14.15.5
* **npm Version:** 7.4.0
* **Operating System:** Mac OS Catalina 10.15.7

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?** Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "env": {
    "node": true,
    "commonjs": true,
    "es2021": true,
    "jest": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "ecmaVersion": 12
  },
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function addOne(a, b, c, d, e) {
  return c + 1;
}

addOne(1, 2, 3, 4, 5);
```

Ran with command
```bash
node_modules/.bin/eslint adder.js
```

**What did you expect to happen?**
I expected an error
```
  1:26  error  'a' is defined but never used  no-unused-vars
  1:29  error  'b' is defined but never used  no-unused-vars
  1:26  error  'd' is defined but never used  no-unused-vars
  1:29  error  'e' is defined but never used  no-unused-vars
```

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
```
  1:26  error  'd' is defined but never used  no-unused-vars
  1:29  error  'e' is defined but never used  no-unused-vars
```

**Steps to reproduce this issue:**

1. 
1. 
1. 

**Are you willing to submit a pull request to fix this bug?**

**Comment by armandAkop** [comment]
@mdjermanovic What rule should I be using to catch this?

**Comment by mdjermanovic** [comment]
Hi @armandAkop, thanks for the issue!

This works as intended, since the default value for the [args](https://eslint.org/docs/rules/no-unused-vars#args) option is [after-used](https://eslint.org/docs/rules/no-unused-vars#args-after-used).

In your example, `c` is used, so the rule doesn't report `a` and `b` because they're before `c`.

You can set ["args": "all"](https://eslint.org/docs/rules/no-unused-vars#args-all) to get warnings for all unused parameters.

[Online Demo with "args": "all"](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFtcImVycm9yXCIsIHsgXCJhcmdzXCI6IFwiYWxsXCIgfV0qL1xuXG5mdW5jdGlvbiBhZGRPbmUoYSwgYiwgYywgZCwgZSkge1xuICByZXR1cm4gYyArIDE7XG59XG5cbmFkZE9uZSgxLCAyLCAzLCA0LCA1KTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by armandAkop** [comment]
Ahh okay, thank you!
